core/src/main/java/org/apache/iceberg/jdbc/JdbcClientPool.java

```diff
@@ -43,8 +53,16 @@ public JdbcClientPool(String dbUrl, Map<String, String> props) {
   }
 
   public JdbcClientPool(int poolSize, String dbUrl, Map<String, String> props) {
-    super(poolSize, SQLNonTransientConnectionException.class, true);
+    super(poolSize, SQLTransientException.class, true);
```

Comment:
using `SQLTransientException` here looks correct to me

Reply:
Yup! I think it was just a bug since the beginning, figured we may as well change the default here since we're addressing retries in generla.